src/main/java/io/reactivex/rxjava3/internal/operators/maybe/MaybeZipArray.java

```diff
@@ -115,6 +118,7 @@ void innerSuccess(T value, int index) {
                     return;
                 }
 
+                Arrays.fill(values, null);
```

Comment:
This should also be in the catch block above, between the `throwIfFatal` and `onError` calls. Similarly, `innerError` should be updated.